File: SWEA1232. [SW 문제해결 기본] 9일차 - 사칙연산/최원재.py
class Node:
    value = -1;
    left = -1;
    right = -1;
    def get_value(self):
        if(self.value == '+'):
            return node_list[int(self.left)].get_value() + node_list[int(self.right)].get_value()
        elif(self.value == '-'):
            return node_list[int(self.left)].get_value() - node_list[int(self.right)].get_value()
        elif(self.value == '*'):
            return node_list[int(self.left)].get_value() * node_list[int(self.right)].get_value()
        elif(self.value == '/'):
            return node_list[int(self.left)].get_value() / node_list[int(self.right)].get_value()
        else:
            return int(self.value)
 
for test_case in range(1, 11):
    n = int(input())
    node_list = [Node() for j in range(n+1)]
    for i in range(1, n+1):
        v = input().split()
        node_list[int(v[0])].value = v[1]
        if v[1] == '+' or v[1] == '-' or v[1] == '*' or v[1] == '/':
            node_list[i].left = v[2]
            node_list[i].right = v[3]
    print("#",test_case," ",int(node_list[1].get_value()),sep="")

File: SWEA1232. [SW 문제해결 기본] 9일차 - 사칙연산/최원재.java
import java.util.Scanner;
import java.io.FileInputStream;
 
class Node{
    char oper = ' ';
    double value = -1;
    Node left, right;
     
    public double get_value() {
        if(oper == '+') {
            return this.left.get_value() + this.right.get_value();
        }
        else if (oper == '-') {
            return this.left.get_value() - this.right.get_value();
        }
        else if (oper == '*') {
            return this.left.get_value() * this.right.get_value();
        }
        else if (oper == '/') {
            return this.left.get_value() / this.right.get_value();
        }
        else {
            return this.value;
        }
    }
}
     
class Solution
{
    public static void main(String args[]) throws Exception
    {
        Scanner sc = new Scanner(System.in);
 
        for(int test_case = 1; test_case <= 10; test_case++)
        {
            int n = sc.nextInt();
            Node[] arr = new Node[n+1];
            for (int i = 1; i<=n ;i++) {
                arr[i] = new Node();
            }
            int curr;
            String line;
            char val;
            for(int i = 1; i <= n; ++i) {
                curr = sc.nextInt();
                line = sc.next();
                val = line.charAt(0);
                if(val == '+' || val == '-' || val == '*' || val == '/' ) {
                    arr[i].left = arr[sc.nextInt()];
                    arr[i].right = arr[sc.nextInt()];
                    arr[i].oper = val;
                }
                else {
                    arr[i].value = Integer.parseInt(line);
                }
            }
            System.out.println("#" + test_case + " " + (int)arr[1].get_value());
        }
    }
}
